src/pubtools/_pulp/tasks/delete.py

```diff
@@ -563,8 +567,17 @@ def log_missing_units(self, searched_units, unit_attr, unit_type, unit_names):
     def map_to_repo(self, units, repos, unit_attr):
         repo_map = {}
         unit_map = {}
+        if ALL_REPOS_INDICATOR in repos:
+            repos = repos.copy()
+            for unit in sorted(units):
```

Comment:
I'm just realizing the negative impact of expanding `*` here, once, covering all units - rather than expanding it every time, *per unit*, in the later loop.

That seems to mean `*`, rather than meaning "Every repo containing the unit being deleted", means "Every repo containing *any* unit in the set of units being deleted".

It doesn't quite seem right, but it doesn't seem like it will majorly break anything either.  It means there will be requests to delete units from repos which already don't contain the unit.  Pointless but not particularly harmful.

Does it produce a bunch of warning spam though, i.e. the "(unit) is not present in (repo)" log in the upcoming loop?

The scenario would be something like:

- unit_a is in repo_a
- unit_b is in repo_b

If user does `delete unit_a --repo *`, followed by `delete unit_b --repo *`, everything is fine.
But if user does `delete unit_a unit_b --repo *`, then it will warn "unit_a is not present in repo_b" and "unit_b is not present in repo_a", won't it?  That does not seem to warrant a warning.

If we want `*` to have the meaning "every repo containing the unit being deleted", then I don't see a way to avoid expanding the repo list separately for each unit.